### PR TITLE
cc_modules: set default meta frequency value when no config available

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
Each cloud-config module defines meta["frequency"] module attribute.
This frequency will be preferred unless overrides are given in
/etc/cloud/cloud.cfg module definitions.
```
## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
Additional test if you want it. It's only checking logs, but hopefully our logs aren't lying to us.

```diff
diff --git a/tests/integration_tests/modules/test_combined.py b/tests/integration_tests/modules/test_combined.py
index eb45062e..58262ce1 100644
--- a/tests/integration_tests/modules/test_combined.py
+++ b/tests/integration_tests/modules/test_combined.py
@@ -5,12 +5,16 @@ of the test would be unlikely to affect the running of another test using
 the same instance launch. Most independent module coherence tests can go
 here.
 """
+import glob
+import importlib
 import json
 import re
 import uuid
+from pathlib import Path
 
 import pytest
 
+import cloudinit.config
 from tests.integration_tests.clouds import ImageSpecification
 from tests.integration_tests.decorators import retry
 from tests.integration_tests.instances import IntegrationInstance
@@ -223,6 +227,25 @@ class TestCombined:
             class_client.execute("stat -c %N /run/cloud-init/cloud-id")
         )
 
+    def test_run_frequency(self, class_client: IntegrationInstance):
+        log = class_client.read_from_file("/var/log/cloud-init.log")
+        config_dir = Path(cloudinit.config.__file__).parent
+        module_paths = glob.glob(str(config_dir / "cc*.py"))
+        module_names = [Path(x).stem for x in module_paths]
+        found_count = 0
+        for name in module_names:
+            mod = importlib.import_module(f"cloudinit.config.{name}")
+            frequency = mod.meta["frequency"]
+            # cc_ gets replaced with config- in logs
+            log_name = name.replace("cc_", "config-")
+            # Some modules have been filtered out in /etc/cloud/cloud.cfg,
+            if f"running {log_name}" in log:
+                found_count += 1  # Ensure we're matching on the right text
+                assert f"running {log_name} with frequency {frequency}" in log
+        assert (
+            found_count > 10
+        ), "Not enough modules found in log. Did the log message change?"
+
     def _check_common_metadata(self, data):
         assert data["base64_encoded_keys"] == []
         assert data["merged_cfg"] == "redacted for non-root user"
```

**Comment by blackboxsw** [comment]
> Additional test if you want it. It's only checking logs, but hopefully our logs aren't lying to us.
> 
> ```diff
> diff --git a/tests/integration_tests/modules/test_combined.py b/tests/integration_tests/modules/test_combined.py
> index eb45062e..58262ce1 100644
> --- a/tests/integration_tests/modules/test_combined.py
> +++ b/tests/integration_tests/modules/test_combined.py
> @@ -5,12 +5,16 @@ of the test would be unlikely to affect the running of another test using
>  the same instance launch. Most independent module coherence tests can go
>  here.
>  """
> +import glob
> +import importlib
>  import json
>  import re
>  import uuid
> +from pathlib import Path
>  
>  import pytest
>  
> +import cloudinit.config
>  from tests.integration_tests.clouds import ImageSpecification
>  from tests.integration_tests.decorators import retry
>  from tests.integration_tests.instances import IntegrationInstance
> @@ -223,6 +227,25 @@ class TestCombined:
>              class_client.execute("stat -c %N /run/cloud-init/cloud-id")
>          )
>  
> +    def test_run_frequency(self, class_client: IntegrationInstance):
> +        log = class_client.read_from_file("/var/log/cloud-init.log")
> +        config_dir = Path(cloudinit.config.__file__).parent
> +        module_paths = glob.glob(str(config_dir / "cc*.py"))
> +        module_names = [Path(x).stem for x in module_paths]
> +        found_count = 0
> +        for name in module_names:
> +            mod = importlib.import_module(f"cloudinit.config.{name}")
> +            frequency = mod.meta["frequency"]
> +            # cc_ gets replaced with config- in logs
> +            log_name = name.replace("cc_", "config-")
> +            # Some modules have been filtered out in /etc/cloud/cloud.cfg,
> +            if f"running {log_name}" in log:
> +                found_count += 1  # Ensure we're matching on the right text
> +                assert f"running {log_name} with frequency {frequency}" in log
> +        assert (
> +            found_count > 10
> +        ), "Not enough modules found in log. Did the log message change?"
> +
>      def _check_common_metadata(self, data):
>          assert data["base64_encoded_keys"] == []
>          assert data["merged_cfg"] == "redacted for non-root user"
> ```

+1 , I was adding something similar to unit tests, but also just a general check that we didn't have anything fall through the cracks with a "None" frequency.
I'll pull in yours as I'd appreciate the full integration test on this, maybe we can suppplement with a no None frequency seen.

File: cloudinit/config/modules.py [hunk 1]
```diff
@@ -186,6 +187,9 @@ def _fixup_modules(self, raw_mods) -> List[ModuleDetails]:
                 continue
             mod = importer.import_module(mod_locs[0])
             validate_module(mod, raw_name)
+            if freq is None:
```
Comment (TheRealFalcon) [inline cloudinit/config/modules.py:190]: Whoops...that was a pretty big miss on my part :grimacing:

**Review by TheRealFalcon — APPROVED** [review]
Thanks! Looks good to me as is. I recently commented another integration test you could add for overriding frequency if you'd like.

File: cloudinit/config/modules.py (hunk 1 above)
Comment (blackboxsw) [reply cloudinit/config/modules.py:190]: My fault in reviewing :)
